addons/binding/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/http/DenonMarantzHttpConnector.java

```diff
@@ -0,0 +1,399 @@
+                httpClient.stop();
+            } catch (Exception e) {
+                logger.debug("Error stopping http client", e);
+            }
+        }
+    }
+
+    /**
+     * Gets the current state of all properties from the receiver, including
+     * basic configuration info (like the number of zones)
+     *
+     * @throws IOException
+     */
+    private void refreshState() {
+        setConfigProperties();
+        refreshHttpProperties();
+    }
+
+    @Override
+    protected void internalSendCommand(String command) {
+        logger.debug("Sending command '{}'", command);
+        if (StringUtils.isBlank(command)) {
+            logger.warn("Trying to send empty command");
+            return;
+        }
+
+        try {
+            String url = cmdUrl + URLEncoder.encode(command, Charset.defaultCharset().displayName());
+            logger.trace("Calling url {}", url);
+
+            if (!httpClient.isStarted()) {
+                httpClient.start();
+            }
+
+            httpClient.newRequest(url).timeout(5, TimeUnit.SECONDS).send(new Response.CompleteListener() {
+                @Override
+                public void onComplete(Result result) {
+                    if (result.getResponse().getStatus() != 200) {
+                        logger.warn("Error {} while sending command", result.getResponse().getReason());
+                    }
+                }
+            });
+
+        } catch (UnsupportedEncodingException e) {
+            logger.warn("Error sending command", e);
+        } catch (Exception e) {
+            logger.warn("Could not start HTTP client", e);
+        }
+    }
+
+    private void updateMain() {
+        String url = statusUrl + URL_MAIN;
+        logger.trace("Refreshing URL: {}", url);
+
+        Main statusMain = getDocument(url, Main.class);
+        if (statusMain != null) {
+            state.setPower(statusMain.getPower().getValue());
+        }
+    }
+
+    private void updateMainZone() {
+        String url = statusUrl + URL_ZONE_MAIN;
+        logger.trace("Refreshing URL: {}", url);
+
+        ZoneStatus mainZone = getDocument(url, ZoneStatus.class);
+        if (mainZone != null) {
+            state.setInput(mainZone.getInputFuncSelect().getValue());
+            state.setMainVolume(mainZone.getMasterVolume().getValue());
+            state.setMainZonePower(mainZone.getPower().getValue());
+            state.setMute(mainZone.getMute().getValue());
+
+            if (config.inputOptions == null) {
+                config.inputOptions = mainZone.getInputFuncList();
+            }
+
+            if (mainZone.getSurrMode() == null) {
+                logger.debug("Unable to get the SURROUND_MODE. MainZone update may not be correct.");
+            } else {
+                state.setSurroundProgram(mainZone.getSurrMode().getValue());
+            }
+        }
+    }
+
+    private void updateSecondaryZones() {
+        for (int i = 2; i <= config.getZoneCount(); i++) {
```

Comment:
Why are you using a loop at all if you have a switch-case later on anyway?

Reply:
To reuse the code to construct the url, to get the document and to log about that.